hapi/hedera-protobuf-java-api/src/main/proto/block/block_service.proto

```diff
@@ -242,26 +227,78 @@ message PublishStreamResponse {
          * <p>
-         * If a source system receives acknowledgement with this flag set
-         * true the source system SHOULD end the stream. The `block_number`
-         * returned SHALL be the last block known and verified by the receiving
-         * system, and the source system MAY resume publishing immediately
-         * after that block.
+         * If a Publisher receives acknowledgement with this flag set
+         * true the Publisher MAY end the stream.<br/>
+         * The `block_number` returned SHALL be the last block known and
+         * verified by the Node.<br/>
+         * The Publisher MAY resume publishing immediately after the indicated
+         * block.
          */
         bool block_already_exists = 3;
     }
 
+    /**
+     * Message indicating the Publisher should skip the current block.
+     *
+     * Nodes SHOULD only skip a block if that block is currently being
+     * received from another source.<br/>
+     * Publishers MUST stop sending the current block and resume with
+     * the block header for the next block.<br/>
+     * A Publisher that receives this message SHALL subsequently receive an
+     * acknowledgement for the skipped block or SHALL receive a `ResendBlock`
+     * message.<br/>
+     * A Publisher asked to skip a block SHOULD NOT delay any subsequent
+     * block, but should send that block as soon as it is available.<br/>
+     * A Publisher MAY be asked to skip multiple blocks in succession, but
+     * SHOULD interpret that as the presence of another Publisher with a
+     * lower latency connection.
+     */
+    message SkipBlock {
+        /**
+         * The number of the _unverified_ block to skip.
+         * <p>
+         * This MUST match the block number of the `BlockHeader` most recently
+         * sent by the Publisher.
+         */
+        uint64 block_number = 2;
+    }
+
+    /**
+     * Message indicating the Publisher must resend from a specified block.
+     * A Publisher might resend the block, if it has that block available,
+     * or it might choose to end the stream if it has already received
+     * acknowledgement from a different trustworthy Node.
+     *
+     * On receiving this message, a Publisher MUST resume sending blocks at
+     * the block number indicated, or else end the stream.<br/>
+     * This message SHALL only be sent following a `SkipBlock` message for a
+     * block that has not been acknowledged.<br/>
+     * A Node SHALL NOT send a `ResendBlock` message for any block that
+     * is already acknowledged.<br/>
+     * A Publisher MAY choose to end the stream rather than resend the
+     * the requested block.
+     */
+    message ResendBlock {
+        /**
+         * The number of the _unverified_ block to re-send.
+         * <p>
+         * This SHALL be the block number one greater than the last block
+         * this Node has successfully stored and verified.
+         */
+        uint64 block_number = 2;
```

Comment:
Similar question should we ever consider a range of blocks to resend or always only 1 at a time?